hw2/05-form.js

```diff
@@ -1 +1,43 @@
-// Add your code here
+let form = document.querySelector("form");
+let registration = document.getElementById("registrationStatus");
+let checkbox1 = document.getElementById("programmingLanguages");
+let checkbox2 = document.getElementById("operatingSystems");
+let checkbox3 = document.getElementById("fullStack");
+
+form.addEventListener("submit", (event) => {
+  console.group();
+  let name = form.inputName.value.length > 0 ? form.inputName.value : "None";
+  console.log("Full name: ", name);
+
+  let email = form.inputEmail.value.length > 0 ? form.inputEmail.value : "None";
+  console.log("Email: ", email);
+
+  let registrationValue = registration.options[registration.selectedIndex].text;
+  console.log("Registration Status: ", registrationValue);
+
+  let selectedCourses = [];
+  if (checkbox1.checked) {
+    selectedCourses.push("Programming Languages");
+  }
+  if (checkbox2.checked) {
+    selectedCourses.push("Operating Systems");
+  }
+  if (checkbox3.checked) {
+    selectedCourses.push("Full Stack Web Development");
+  }
```

Comment:
This looks great,  but the checkbox logic can be condensed quite a bit by grabbing all courses at once. This was my solution:

let checkboxes = document.getElementsByName("Courses");
  let courses = "Courses: ";
  for (let i = 0; i < checkboxes.length; i++) {
    if (checkboxes[i].checked) {
      courses += checkboxes[i].value + ", ";
    }
  }
  console.log(courses);
  
  Excellent work all around, everything looks great!

Reply:
Thanks for the suggestion Steven! I really appreciate it. I initially tried displaying the checkbox outputs using a similar `for loop` as you have mentioned, but one small catch I got into was if the user checks a single subject then also in the console it displays with a "**`,`**". Say for example: programming languages**,**

Reply:
Thanks for the suggestion Steven! I really appreciate it. I will try to convert my 3 `if `statements into a single `for `loop and update my code.

Reply:
If you wanted, there is a way to get the boxes that are checked with the querySectorAll() method so you don't need to make sure they are checked yourself.

Reply:
Thanks for the feedback Corey. I have learned something new today and I have made those changes in my last commit.